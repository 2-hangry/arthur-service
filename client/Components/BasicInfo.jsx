import React from "react";
import ReviewHealth from "./ReviewHealth.jsx";
import PriceRange from "./PriceRange.jsx";
const moment = require("moment");

const imageStyle = { width: 25, height: 25 };

const textStyle = { fontSize: 14, fontFamily: "sans-serif" };

const hoursTodayStyling = { marginTop: 7, marginLeft: 5 };

const BasicInfo = props => {
  const time = props.businessHours.open[moment().format("d") - 1];
  const start = moment(Math.floor(time.start) / 100, "HH").format("hh:mm a");
  const end = moment(Math.floor(time.end) / 100, "HH").format("hh:mm a");
  let current = moment().format("hh:mm a");
  return (
    <div style={textStyle}>
      <span style={{ display: "flex" }}>
        <img
          src="https://png.icons8.com/metro/50/27ae60/clock.png"
          style={imageStyle}
        />
        <p style={hoursTodayStyling}>Today</p>
        <p style={Object.assign({ fontWeight: "bold" }, hoursTodayStyling)}>
          {start} - {end}
        </p>
        {end > current && current < end ? (
          <p
            style={Object.assign(
              {
                fontWeight: "bold",
                color: "green"
              },
              hoursTodayStyling
            )}
          >
            {" "}
            Open Now
          </p>
        ) : (
          <p
            style={Object.assign(
              {
                fontWeight: "bold",
                color: "red"
              },
              hoursTodayStyling
            )}
          >
            {" "}
            Closed Now
          </p>
        )}
      </span>
      <PriceRange price={props.price} />
      <ReviewHealth rating={props.rating} />
      <a href="https://icons8.com" />
    </div>
  );
};

export default BasicInfo;
