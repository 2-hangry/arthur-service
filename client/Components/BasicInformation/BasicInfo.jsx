import React from 'react';
import PropTypes from 'prop-types';
import ReviewHealth from './ReviewHealth';
import PriceRange from './PriceRange';
import styles from '../../styles.css';

const moment = require('moment');

const BasicInfo = (props) => {
  const { businessHours, price, rating } = props;
  const time = businessHours.open[moment().format('d') - 1];
  const start = moment(Math.floor(time.start) / 100, 'HH').format('hh:mm a');
  const end = moment(Math.floor(time.end) / 100, 'HH').format('hh:mm a');
  const current = moment().format('hh:mm a');

  return (
    <div>
      <span style={{ display: 'flex' }}>
        {moment(current, 'hh:mm a').isBetween(moment(start, 'hh:mm a'), moment(end, 'hh:mm a')) ? (
          <img
            src="https://png.icons8.com/metro/50/27ae60/clock.png"
            alt="Clock"
            style={styles.clockStyling}
          />
        ) : (
          <img
            src="https://png.icons8.com/metro/50/fc0000/clock.png"
            alt="Clock"
            style={styles.clockStyling}
          />
        )}
        <p style={styles.hoursTodayStyling}> Today</p>
        <p style={Object.assign({ fontWeight: 'bold' }, styles.hoursTodayStyling)}>
          {start}
          {' '}
-
          {end}
        </p>
        {moment(current, 'hh:mm a').isBetween(moment(start, 'hh:mm a'), moment(end, 'hh:mm a')) ? (
          <p
            style={Object.assign({ fontWeight: 'bold', color: 'green' }, styles.hoursTodayStyling)}
          >
            {' '}
            Open Now
          </p>
        ) : (
          <p style={Object.assign({ fontWeight: 'bold', color: 'red' }, styles.hoursTodayStyling)}>
            {' '}
            Closed Now
          </p>
        )}
      </span>
      <hr style={styles.horizontalRuleStyling} />
      <PriceRange price={price} />
      <hr style={styles.horizontalRuleStyling} />
      <ReviewHealth rating={rating} />
    </div>
  );
};

export default BasicInfo;
