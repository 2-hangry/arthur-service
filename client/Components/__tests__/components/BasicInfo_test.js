import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import BasicInfo from "../../BasicInformation/BasicInfo.jsx";

Enzyme.configure({ adapter: new Adapter() });
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const props = {
    businessHours: {
      open: [
        {
          day: 0,
          end: 2014,
          start: 897
        },
        {
          day: 1,
          end: 2100,
          start: 700
        },
        {
          day: 2,
          end: 2300,
          start: 700
        },
        {
          day: 3,
          end: 1600,
          start: 300
        },
        {
          day: 4,
          end: 2100,
          start: 700
        },
        {
          day: 5,
          end: 2300,
          start: 700
        },
        {
          day: 6,
          end: 1600,
          start: 300
        }
      ]
    },
    price: "$$$$",
    rating: 3.1
  };
  const tree = renderer.create(<BasicInfo {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
