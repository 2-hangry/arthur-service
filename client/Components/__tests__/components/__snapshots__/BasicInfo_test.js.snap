// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <span
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <img
      src="https://png.icons8.com/metro/50/27ae60/clock.png"
      style={
        Object {
          "height": 28,
          "width": 28,
        }
      }
    />
    <p
      style={
        Object {
          "marginLeft": 5,
          "marginTop": 7,
        }
      }
    >
       
      Today
    </p>
    <p
      style={
        Object {
          "fontWeight": "bold",
          "marginLeft": 5,
          "marginTop": 7,
        }
      }
    >
      07:00 am
       - 
      11:00 pm
    </p>
    <p
      style={
        Object {
          "color": "green",
          "fontWeight": "bold",
          "marginLeft": 5,
          "marginTop": 7,
        }
      }
    >
       
      Open Now
    </p>
  </span>
  <hr
    style={
      Object {
        "marginLeft": "11%",
        "marginRight": "2%",
      }
    }
  />
  <div>
    <span
      style={
        Object {
          "display": "flex",
        }
      }
    >
      <img
        src="https://png.icons8.com/metro/50/27ae60/us-dollar.png"
        style={
          Object {
            "height": 18,
            "mode": "fit",
            "width": 17,
          }
        }
      />
      <img
        src="https://png.icons8.com/metro/50/27ae60/us-dollar.png"
        style={
          Object {
            "height": 18,
            "mode": "fit",
            "width": 17,
          }
        }
      />
      <img
        src="https://png.icons8.com/metro/50/27ae60/us-dollar.png"
        style={
          Object {
            "height": 18,
            "mode": "fit",
            "width": 17,
          }
        }
      />
      <img
        src="https://png.icons8.com/metro/50/27ae60/us-dollar.png"
        style={
          Object {
            "height": 18,
            "mode": "fit",
            "width": 17,
          }
        }
      />
      <p
        style={
          Object {
            "marginLeft": 6,
            "marginTop": 4,
          }
        }
      >
        Price range
      </p>
      <p
        style={
          Object {
            "fontWeight": "bold",
            "marginLeft": 6,
            "marginTop": 4,
          }
        }
      >
        above $61
      </p>
    </span>
    <a
      href="https://icons8.com"
    />
  </div>
  <hr
    style={
      Object {
        "marginLeft": "11%",
        "marginRight": "2%",
      }
    }
  />
  <div>
    <a
      onClick={[Function]}
    >
      Review Health
    </a>
  </div>
  <a
    href="https://icons8.com"
  />
</div>
`;
